src/components/InputBox.js

```diff
@@ -1,166 +1,150 @@
 import './Input.css';
-import InputForm from "./forms/input";
-import GenericDropdown from "./forms/genericdropdown";
-import {ancestries, ancestMap} from "../constants/ancestries";
-import equipment from "../constants/statics/equipment";
-import experience from "../constants/experience";
-import type from "../constants/statics/type";
-import {races, raceMap} from '../constants/races';
-import {traits, traitMap} from '../constants/traits';
-import { useRef, useState } from "react";
+import InputForm from './forms/input';
+import GenericDropdown from './forms/genericdropdown';
+import { ancestries, ancestMap } from '../constants/ancestries';
+import equipment from '../constants/statics/equipment';
+import experience from '../constants/experience';
+import type from '../constants/statics/type';
+import { races, raceMap } from '../constants/races';
+import { traits, traitMap } from '../constants/traits';
+import { useRef, useState, useEffect, useCallback, useMemo } from 'react';
 import defaults from '../constants/statics/defaults';
 import ArrayReadingDropdown from './forms/arrayreadingdropdown';
 import SelectiveDropdown from './forms/selectivedropdown';
 
-const InputBox = ({onMod}) => {
-    const [isLevy, setLevy] = useState(false);
-    const [race, setRace] = useState(raceMap.get(defaults.race));
-    const [traitList, setTraitList] = useState(defaults.traits);
-    const [ancest, setAncest] = useState(defaults.ancestry);
+const InputBox = ({ onMod }) => {
+	const [isLevy, setLevy] = useState(false);
+	const [race, setRace] = useState(raceMap.get(defaults.race));
+	const [traitList, setTraitList] = useState(defaults.traits);
+	const [ancest, setAncest] = useState(defaults.ancestry);
 
-    const nameRef = useRef(defaults.name);
-    const commandRef = useRef();
-    const ancestRef = useRef(defaults.ancestry);
-    const typeRef = useRef(defaults.type);
-    const expRef = useRef(defaults.exp);
-    const equipRef = useRef(defaults.equip);
-    const raceRef = useRef(defaults.race);
-    const traitRef = useRef();
-    var list = [];
+	const nameRef = useRef(defaults.name);
+	const commandRef = useRef();
+	const ancestRef = useRef(defaults.ancestry);
+	const typeRef = useRef(defaults.type);
+	const expRef = useRef(defaults.exp);
+	const equipRef = useRef(defaults.equip);
+	const raceRef = useRef(defaults.race);
+	const traitRef = useRef();
+	let list = useMemo(() => [], [traitList]);
```

Comment:
```suggestion
	let list = useMemo(() => [], []);
```

Reply:
This is the only thing I'm worried about. It has no dependency, so I think it's rendering infinitely??????